core/common/src/main/java/alluxio/conf/InstancedConfiguration.java

```diff
@@ -336,14 +337,31 @@ public void validate() {
     if (!getBoolean(PropertyKey.CONF_VALIDATION_ENABLED)) {
       return;
     }
+    List<String> removalMessages = new ArrayList<>();
     for (PropertyKey key : keySet()) {
       Preconditions.checkState(
           getSource(key).getType() != Source.Type.SITE_PROPERTY || !key.isIgnoredSiteProperty(),
           "%s is not accepted in alluxio-site.properties, "
               + "and must be specified as a JVM property. "
               + "If no JVM property is present, Alluxio will use default value '%s'.",
           key.getName(), key.getDefaultValue());
+
+      if (PropertyKey.isDeprecated(key) && getSource(key).compareTo(Source.DEFAULT) > 0) {
+        LOG.warn("{} is deprecated. Please avoid using this key in the future. {}", key.getName(),
+            PropertyKey.getDeprecationMessage(key));
+      } else if (PropertyKey.isRemoved(key)) {
+        String errorMsg = String.format("%s is no longer a valid property. %s", key.getName(),
+            PropertyKey.getRemovalMessage(key));
+        removalMessages.add(errorMsg);
+        LOG.error(errorMsg);
+      }
     }
+
+    if (removalMessages.size() > 0) {
+      String msg = removalMessages.stream().reduce((m1, m2) -> m1 + "\n" + m2).get();
```

Comment:
`Collectors.joining()` is more idiomatic and efficient

Reply:
I put `Collectors.joining` and my IDE warned me about the function utilizing too many temporary objects. It suggested using `String.join` as a more efficient alternative instead

```java
String.join("\n", removalMessages);
```